### PR TITLE
SAAS-490 SAAS-507 user profile page + modify user's name

### PR DESCRIPTION
**Pull request by nicolalamacchia**

### CODE REVIEW

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts [hunk 1]
```diff
@@ -235,5 +235,85 @@ export class AuthAPIClient {
     this.methodInfoResetPassword);
   }
 
+  methodInfoGetProfile = new grpcWeb.AbstractClientBase.MethodInfo(
+    auth_auth_api_pb.GetProfileResponse,
+    (request: auth_auth_api_pb.GetProfileRequest) => {
+      return request.serializeBinary();
```
Comment (fabio-silva) [inline src/core/gen/web/auth/Auth_apiServiceClientPb.ts:241]: nit: immediate return

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts [hunk 2]
```diff
@@ -235,5 +235,85 @@ export class AuthAPIClient {
...
+               response: auth_auth_api_pb.GetProfileResponse) => void): grpcWeb.ClientReadableStream<auth_auth_api_pb.GetProfileResponse>;
+
+  getProfile(
+    request: auth_auth_api_pb.GetProfileRequest,
+    metadata: grpcWeb.Metadata | null,
+    callback?: (err: grpcWeb.Error,
+               response: auth_auth_api_pb.GetProfileResponse) => void) {
+    if (callback !== undefined) {
+      return this.client_.rpcCall(
+        this.hostname_ +
```
Comment (fabio-silva) [inline src/core/gen/web/auth/Auth_apiServiceClientPb.ts:263]: nit: string template

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts [hunk 3]
```diff
@@ -235,5 +235,85 @@ export class AuthAPIClient {
...
+          '/percona.platform.auth.v1.AuthAPI/GetProfile',
+        request,
+        metadata || {},
+        this.methodInfoGetProfile,
+        callback);
+    }
+    return this.client_.unaryCall(
+    this.hostname_ +
```
Comment (fabio-silva) [inline src/core/gen/web/auth/Auth_apiServiceClientPb.ts:271]: nit: same as above

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts (hunk 1 above)
Comment (fabio-silva) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:241]: also there are similar patterns below

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts [hunk 4]
```diff
@@ -235,5 +235,85 @@ export class AuthAPIClient {
...
+        request,
+        metadata || {},
+        this.methodInfoUpdateProfile,
+        callback);
+    }
+    return this.client_.unaryCall(
+    this.hostname_ +
+      '/percona.platform.auth.v1.AuthAPI/UpdateProfile',
+    request,
+    metadata || {},
```
Comment (fabio-silva) [inline src/core/gen/web/auth/Auth_apiServiceClientPb.ts:314]: nit: if this is to prevent null/undefined , nullish coalescing operator could be used

Comment (fabio-silva) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:314]: (also present in other parts in this code chunk)

File: src/pages/Profile/Profile.styles.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      font-size: ${typography.heading.h3};
+      font-weight: ${typography.weight.regular};
+      margin: ${spacing.formMargin};
+      text-align: center;
+      color: ${colors.text};
+    `,
+    nameFields: css`
+      display: flex;
+
+      & > * {
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.styles.ts:26]: I can tell we're targeting `TextInputField`s down there. Is there a better option that `*` ?

**Review by fabio-silva — APPROVED** [review]
Just some nits, nothing much aside from that 👌

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts [hunk 6]
```diff
@@ -235,5 +235,85 @@ export class AuthAPIClient {
...
+
+  getProfile(
+    request: auth_auth_api_pb.GetProfileRequest,
+    metadata: grpcWeb.Metadata | null,
+    callback?: (err: grpcWeb.Error,
+               response: auth_auth_api_pb.GetProfileResponse) => void) {
+    if (callback !== undefined) {
+      return this.client_.rpcCall(
+        this.hostname_ +
+          '/percona.platform.auth.v1.AuthAPI/GetProfile',
```
[the remark this answers is not in the document]
Comment (atymchuk) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:264]: oops, it's generated, so not much we can do )

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts (hunk 1 above)
Comment (nicolalamacchia) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:241]: this is generated code

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts (hunk 2 above)
Comment (nicolalamacchia) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:263]: this is generated code

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts (hunk 3 above)
Comment (nicolalamacchia) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:271]: this is generated code

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts (hunk 4 above)
Comment (nicolalamacchia) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:314]: this is generated code

File: src/components/MenuBar/MenuBar.tsx [hunk 7]
```diff
@@ -20,6 +21,10 @@ export const MenuBar: FC = () => {
   const { authenticated, email } = useSelector(getAuth);
   const currentTheme = useSelector(getCurrentTheme);
 
+  const goToProfilePage = async () => {
```
[the remark this answers is not in the document]
Comment (nicolalamacchia) [reply src/components/MenuBar/MenuBar.tsx]: done

File: src/core/gen/web/auth/Auth_apiServiceClientPb.ts (hunk 4 above)
Comment (fabio-silva) [reply src/core/gen/web/auth/Auth_apiServiceClientPb.ts:314]: oh, ok, didn't know about that

File: src/pages/Profile/Profile.styles.ts (hunk 5 above)
Comment (nicolalamacchia) [reply src/pages/Profile/Profile.styles.ts:26]: `nameFields` is a wrapper around two text input fields, this is targeting those immediate children. There might be, but this should be good enough. What do you suggest?

Comment (fabio-silva) [reply src/pages/Profile/Profile.styles.ts:26]: it's just something to be careful about I guess, but if that wrapper is merely for that purpose, then we can leave it I guess

**Review by fabio-silva — DISMISSED** [review]

File: src/pages/Profile/Profile.styles.ts (hunk 5 above)
Comment (nicolalamacchia) [reply src/pages/Profile/Profile.styles.ts:26]: yes, the wrapper is there to put the two fields on the same line :)

File: src/core/api/__mocks__/auth.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+// export const getBackendSrv = jest.fn().mockReturnValue(backendSrv);
```
Comment (atymchuk) [inline src/core/api/__mocks__/auth.ts:1]: Leftover

Comment (atymchuk) [inline src/core/api/__mocks__/auth.ts:1]:
```suggestion
```

**Review by atymchuk — APPROVED** [review]

File: src/core/api/__mocks__/auth.ts (hunk 8 above)
Comment (nicolalamacchia) [reply src/core/api/__mocks__/auth.ts]: Done

File: src/core/api/auth.ts [hunk 9]
```diff
@@ -4,21 +4,24 @@ import { SignIn, SignUp, UpdateProfile } from './types';
...
   SignOutRequest,
-  GetProfileRequest,
+  // GetProfileRequest,
   UpdateProfileRequest,
 } = AuthPB;
 
-export const refreshSession = () => {
-  const apiClient = new AuthAPIClient(PLATFORM_BASE_URL, null, null);
+export const refreshSession = () =>
+  // const apiClient = new AuthAPIClient(PLATFORM_BASE_URL, null, null);
```
Comment (atymchuk) [inline src/core/api/auth.ts:16]: Consider cleaning up the comments in this file.

File: src/core/api/auth.ts [hunk 10]
```diff
@@ -1,19 +1,27 @@
...
 
-  const request = new RefreshSessionRequest();
+  // const request = new RefreshSessionRequest();
 
-  return apiClient.refreshSession(request, {});
-};
+  // return apiClient.refreshSession(request, {});
```
Comment (atymchuk) [inline src/core/api/auth.ts:20]: It can't be the reason why the e2e tests fail. Let's put it back :)

File: src/core/api/auth.ts [hunk 11]
```diff
@@ -46,3 +54,27 @@ export const signOut = () => {
...
 };
+
+export const getProfile = () =>
+  // const apiClient = new AuthAPIClient(PLATFORM_BASE_URL, null, null);
+
+  // const request = new GetProfileRequest();
+
+  // return apiClient.getProfile(request, {});
+   ({
+    getFirstName: () => 'Nicola',
```
Comment (atymchuk) [inline src/core/api/auth.ts:65]: Must have mistakenly left it here.

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: src/core/api/auth.ts (hunk 11 above)
Comment (nicolalamacchia) [reply src/core/api/auth.ts]: I told myself: "remember to revert this"... :man_facepalming:

File: src/core/api/auth.ts (hunk 10 above)
Comment (nicolalamacchia) [reply src/core/api/auth.ts]: Bad commit, fixed

File: src/core/api/auth.ts (hunk 9 above)
Comment (nicolalamacchia) [reply src/core/api/auth.ts]: Bad commit, fixed

File: src/pages/Profile/Profile.tsx [hunk 12]
```diff
@@ -34,8 +34,8 @@ export const ProfilePage: FC = () => {
             <form name="login-form" data-qa="login-form" className={styles.form} onSubmit={handleSubmit}>
               <legend className={styles.legend}>{Messages.profile}</legend>
               <div className={styles.nameFields}>
-                <TextInputField label={Messages.firstNameLabel} name="firstName" />
-                <TextInputField label={Messages.lastNameLabel} name="lastName" />
+                <TextInputField validate={validators.required} label={Messages.firstNameLabel} name="firstName" />
+                <TextInputField validate={validators.required} label={Messages.lastNameLabel} name="lastName" />
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.tsx:38]: Most of the times I believe we use `validators`, here it's `validate`. Is there a reason for that?

**Review by fabio-silva — DISMISSED** [review]
Just a nit

File: src/pages/Profile/Profile.tsx (hunk 12 above)
Comment (nicolalamacchia) [reply src/pages/Profile/Profile.tsx]: Fixed

**Review by fabio-silva — APPROVED** [review]

File: src/pages/Profile/Profile.tsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+              </div>
+              <div className={styles.emailFieldWrapper}>
+                <TextInputField disabled label={Messages.emailLabel} name="email" />
+                  <a href={Routes.changeEmail} target="_blank" data-qa="profile-change-email-button" className={styles.externalLink} rel="noreferrer">
+                    {Messages.changeEmail}
+                  </a>
+              </div>
+              <div className={styles.buttonWrapper}>
+                <LoaderButton loading={pending} data-qa="profile-submit-button" type="submit" disabled={!valid || pending || pristine}>
+                  Save
```
Comment (tiagomotasantos) [inline src/pages/Profile/Profile.tsx:52]: This "Save" could be in the Messages file like the rest

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

File: src/pages/Profile/Profile.tsx (hunk 13 above)
Comment (nicolalamacchia) [reply src/pages/Profile/Profile.tsx]: Fixed

**Review by tiagomotasantos — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]

**Pull request merged**